server/ai_live_video.go

```diff
@@ -81,14 +82,32 @@ func startTrickleSubscribe(ctx context.Context, url *url.URL, params aiRequestPa
 
 	// read segments from trickle subscription
 	go func() {
+		var err error
 		defer w.Close()
+		defer func() { params.liveParams.errorCallback(err) }() // wrapper function to be able to reference the latest err
+		retries := 0
 		for {
-			segment, err := subscriber.Read()
+			if !params.inputStreamExists() {
+				clog.Errorf(ctx, "Stopping trickle subscribe, input stream does not exist.")
+				break
+			}
+			var segment *http.Response
+			segment, err = subscriber.Read()
```

Comment:
Will this work fine if called multiple times after retries? @j0sh

Reply:
Depending on where it fails - it might try to request the _next_ segment instead of the current one. For our purposes here that is probably fine. Not ideal, but fine for now.

If you really wanted to fix that, we can probably add a Retry() helper to the subscriber that takes a sequence number (or a http.Response and call [GetSeq](https://github.com/livepeer/go-livepeer/blob/master/trickle/trickle_subscriber.go#L40) on it), then temporarily sets the subscriber `idx` , runs (pre-)connect, then resets `idx`, and returns the response. I am doing something similar in an upcoming change set for the publisher side.